src/schema/v2/homeView/mixer/rules/AuctionEngagementRule.ts

```diff
@@ -0,0 +1,55 @@
+import { HomeViewSection } from "schema/v2/homeView/sections"
+import { ResolverContext } from "types/graphql"
+import { HomeViewMixerRule } from "../HomeViewMixerRule"
+import { NewWorksForYou } from "../../sections/NewWorksForYou"
+import { AuctionLotsForYou } from "../../sections/AuctionLotsForYou"
+import { Auctions } from "../../sections/Auctions"
+import { LatestAuctionResults } from "../../sections/LatestAuctionResults"
+import { compact } from "lodash"
+
+/**
+ * Rule that moves auction-related sections near the top for eligible users
+ * based on their auction segmentation.
+ *
+ * See: https://www.notion.so/artsy/19fcab0764a080229129edbd6efa7dce
+ */
+export class AuctionEngagementRule extends HomeViewMixerRule {
+  async apply(
+    sections: HomeViewSection[],
+    context: ResolverContext
+  ): Promise<HomeViewSection[]> {
+    const segment = await getUserAuctionSegmentation(context)
```

Comment:
We should keep an eye on how much it slows down the home screen, since it’s the first time we’re querying anything besides the `user` object from Gravity to re-order sections.